### PR TITLE
LPS-160185 Add type declaration to Collapse component in layout-content-page-editor-web module

### PR DESCRIPTION
**Pull request by danvb112**
Related issue: https://issues.liferay.com/browse/LPS-160185

### Motivation

Hello Reviewer, I'm working on a technical debt to remove some `@ts-ignore` annotations from the Objects code. For this I have to create some type declarations for components that were not developed in `typescript`, I can declare these types in the `object-js-components-web` module, but I think it will be restricted just for our case, so I decided to declare the types in the `layout-content-page-editor-web` which is the root of the `Collapse` component I had to remove the `@ts-ignore`, so anyone who wants to use this component with `typescript` in the future doesn't need to declare the same types in their own module, avoiding code repetition. What do you think?

### CODE REVIEW

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by p2kmgcl** [comment]
Hey @danvb112! Thanks so much for the contribution. Have you considered migrating the Collapse component to TypeScript instead? I have seen that it has no extra dependencies, so it should be easy :smile:

**Comment by danvb112** [comment]
Hey @p2kmgcl, Thanks for the review, it's a great idea to convert the component to typescript I considered migrating the component to typescript but as I didn't know your opinion about it, I preferred to send only the type definition, I'll work on it!

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
Hey @p2kmgcl, I converted the component to Typescript, could you check if everything is ok please?

**Comment by danvb112** [comment]
ci:test:relevant

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+export {default as Collapse} from '../page_editor/common/components/Collapse';
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/index.d.ts:15]: I think this should be index.ts now :smile:

**Comment by p2kmgcl** [comment]
> Hey @p2kmgcl, Thanks for the review, it's a great idea to convert the component to typescript I considered migrating the component to typescript but as I didn't know your opinion about it, I preferred to send only the type definition, I'll work on it! smile

Makes sense. I have been wanting to migrate the page editor to TypeScript for months, but never have the time to start :sweat_smile:. I could not lose this opportunity.

The PR looks almost perfect, I just added a small comment.

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/index.d.ts (hunk 1 above)
Comment (danvb112) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/index.d.ts]: Makes sense, I also moved the new `index.ts` into the `page_editor/common` directory which is where the other `index.js` exists, let me know if make sense to you :smile:

**Comment by danvb112** [comment]
> Makes sense. I have been wanting to migrate the page editor to TypeScript for months, but never have the time to start sweat_smile. I could not lose this opportunity.

Here in the Objects code we use Typescript and it is much easier to develop and use the components :smile: 

> The PR looks almost perfect, I just added a small comment.
> 
> **Edited**: a couple of things:
> 
> * I just realized that the Collapse.js file is still there, shouldn't we remove it?
> * This PR might have some conflicts with this one in Brian's repo (first commit): [brianchandotcom@1173d92](https://github.com/brianchandotcom/liferay-portal/commit/1173d92b08213fc0474eeb5f5c68a7e57fe3ddaf)

1. Here the `Collapse.js` file is showing up as a renamed file, it no longer exists :thinking:
2. Here it is not showing any conflicts, I rebased my brach a few minutes ago, let me know if you still see any conflicts

**Comment by danvb112** [comment]
ci:test:sf

**Comment by p2kmgcl** [comment]
> Here the Collapse.js file is showing up as a renamed file, it no longer exists

My fault, I didn't see that :sweat_smile: 

> Here it is not showing any conflicts, I rebased my brach a few minutes ago, let me know if you still see any conflicts

Wild conflict appeared :imp:

**Comment by danvb112** [comment]
> Wild conflict appeared :imp:

Done! :smile:

**Comment by p2kmgcl** [comment]
Looks perfect :smile:

**Comment by p2kmgcl** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:reevaluate:1327401_5246

**Comment by ealonso** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:test:relevant

**Comment by ozysouza** [comment]
ci:forward

**Comment by ozysouza** [comment]
The failures are unrelated this PR. In case this last forward also does not work I will manually send it.

**Comment by ozysouza** [comment]
Please see https://github.com/brianchandotcom/liferay-portal/pull/121667

**Comment by ozysouza** [comment]
ci:close

**Pull request closed without merging**